x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/api/engines/delete_engines_api_logic.ts

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { i18n } from '@kbn/i18n';
+
+import { Actions, createApiLogic } from '../../../shared/api_logic/create_api_logic';
+import { HttpLogic } from '../../../shared/http';
+
+export interface DeleteEnginesApiLogicArguments {
+  engineName: string;
+}
+export interface DeleteEnginesApiLogicResponse {
+  engineName: string;
+}
+
+export const deleteEngine = async ({
+  engineName,
+}: DeleteEnginesApiLogicArguments): Promise<DeleteEnginesApiLogicResponse> => {
+  const route = `/internal/enterprise_search/engines/${engineName}`;
+  await HttpLogic.values.http.delete<DeleteEnginesApiLogicResponse>(route);
```

Comment:
will this throw an exception if the delete fails? A case where that would happen is if the user lacks permission to delete the engine. I _think_ we would need to check the response and it does not throw an exception, but I'm not sure.

Reply:
I am not very sure about it. But I was thinking that eventually, we might have to list engines based on the user privileges.  

My assumption - If  a user doesn't have manage privileges, we should not should delete option but view engines

Reply:
that is an option, but a delete could fail for other reasons, permission is just one case. So I think we should handle the response from this call and not just assume it succeeded.

Reply:
Agreed. I am thinking of adding a `try/catch` block and display the error may be?

Reply:
You don't need a try/catch block, just to `return await` the result of the API call. If the API return is an error, the API logic will change its `status` to ERROR and contain an error object with the relevant details.

Reply:
Just some thoughts. If delete engine api returns an exception. API logic should redirect ERROR to apiError and should potentially call `flashApiErrors ` ?

Reply:
Doing some testing I'm pretty sure this will throw an exception so this code will work as written.

Reply:
@TattdCodeMonkey Thank you. Is there something that I should do from my end to handle this situation?

Reply:
Thanks @sphilipse for detailed explanation. That answered my question.